### ISSUE TITLE
Getting strange precipitation patterns in Isca

### ISSUE THREAD

**Issue by mcthreems** [ExeClim/Isca#110]
Hello again,

We've recently started getting strange precipitation patterns in Isca, and it's starting to seem like an issue with the model rather than some new settings we've been trying. We initially saw it after adding a variable heat capacity option for bucket depth over land grid cells. However, I recently decided to rerun an older experiment that uses the default static heat capacity calculation and still received strange precipitation patterns. To demonstrate what we're seeing, I'm attaching first the 1-year precipitation from a previous run using the default heat capacity, and then the recent run of the same experiment (values are in mm/day). I'm not sure if you will know what's going on, but thought it best to at least ask.

![shc_y110](https://user-images.githubusercontent.com/42620929/75613205-ea19fd00-5adf-11ea-9683-6e6ecf37552d.png)
![shc_y6](https://user-images.githubusercontent.com/42620929/75613208-f00fde00-5adf-11ea-98b6-259451194d02.png)

**Comment by gkvallis** [comment]
Just to be clear. Are you saying you have run exactly the same experiment again with the model, but the second time you get different and unrealistic results?  Presumably something was different, though?

**Comment by mcthreems** [comment]
Yes, this is supposed to be the exact same experiment. The settings for the experiment are the same, what's not is edits to the code related to other experiments that shouldn't affect this one. However, it seems the edits are affecting it, and I'm unsure how. I wanted to show you guys what I'm getting to see if you might have some insight into what part of the model might be going wrong to produce the strange precipitation patterns. My guess is it's the tam hydrology edits I made while visiting, but I've looked through and couldn't figure out what in that would affect this run.

**Comment by gkvallis** [comment]
We have ‘trip tests’ that pick up things like this. You make some edits and run a sequence of test cases that compare against a previous commit and it will find any unintentional changes in results (and intentional ones too). You might investigate using those.

**Comment by sit23** [comment]
I agree with Geoff that our `trip_test` functionality would be a good way to investigate this. From your plot, I wouldn't be so much concerned by the pattern, but more concerned that it doesn't seem to be raining at all. My guess would be that evaporation isn't happening, and so your atmosphere is only left with the initial (low) value of specific humidity everywhere. It then only rains a tiny bit at the cold poles, which might explain your pattern? I'd check the `flux_lhe` variable to see if evaporation is happening, and I'd also look at `sphum` and `rh` to see if you are indeed getting very little atmospheric moisture. Could it perhaps be something to do with any changes you've made to the `mixed_layer` code that has, perhaps, meant that the `evaporation` variable is effectively `False` now, where it used to be `True`?

**Comment by RuthG** [comment]
In addition to trip tests, if you made multiple Git commits in incorporating your changes, then one option could be to run Isca from those commits (the python scripts include functionality for this), this will help work out at what point the code broke. 
You could also use Github's 'compare' button (or just a diff, but I find that less readable) to make a record of the changes you have made, which might help give an idea of what has gone wrong. e.g. https://github.com/mcthreems/Isca/compare/master...mcthreems:hydrology_dev 
We can offer some guesses of what might have happened (see Stephen's thoughts above), but you are 'the expert' on the changes you have made, and unfortunately this sort of problem is a 'fun' element of model development for research :(  But troubleshooting will get easier with experience!

**Comment by mcthreems** [comment]
I reran the experiment using an older version of the code from before I made any TAM edits, and I still get the very low precipitation values. I took a look at the evaporation output you mentioned, and it does indeed seem very low, although I'm not confident in my ability to gauge the expected value. I've attached an plot of this value below. I've also attached a plot of the specific humidity at the lowest model level. My main confusion now is that I'm getting these odd values even though I'm using code that should be the same as when I got the much more reasonable values before. I'm concerned that something in the cluster we're using itself has changed.

![image](https://user-images.githubusercontent.com/42620929/75808258-a8858e00-5d3b-11ea-9c90-81740073515e.png)
![image](https://user-images.githubusercontent.com/42620929/75808464-fef2cc80-5d3b-11ea-82e2-75930fc1a54e.png)

**Comment by gkvallis** [comment]
If this is indeed due to changes in  your local computer system then that's not really something we  can help with I'm afraid. That would be a little surprising, though, and before going down that route I'd want to make sure it wasn't a code change. You say you are now using the code that 'should be the same' as before. If you have git commits you should be able to ensure you are running _exactly_ the same code.  You might also rerun a couple of the test cases we have posted in Exeter and see if they still give sensible answers on your local computers.

**Comment by mcthreems** [comment]
As far as I understand it should be the same code, I moved to a branch that was made before adding any of the edits for TAM. I ran the socrates aquaplanet test to see if it gave me the expected outputs, and it does seem to have given reasonable values (see below), so I'll have to figure out what in the code could cause this difference.

![image](https://user-images.githubusercontent.com/42620929/75898885-95cc9100-5def-11ea-9172-83ac96eedac3.png)

**Comment by mcthreems** [comment]
I have found the reason for the issue I've been having. I backtracked from the evaporation output, which was indeed much lower in my more recent runs. I found that in the surface_flux module, at one point in the calculation of the q_flux there is a distinction made depending on how the bucket depth compares to the maximum depth. I assume this is normally done to simulate liquid that is at or near the surface of a land cell, but for our purposes it resulted in very low evaporation. This is because I had recently changed the max bucket depth to be very large, such that we would not lose any water from the system. I added a flag to my version of the code that avoids this distinction if turned on, such that my "land" grid cells are treated like ocean cells in for that specific calculation. I am now getting much more reasonable precipitation values, so I will considered this issue solved for the time being.
